remix/app/repositories/user.server.ts

```diff
@@ -0,0 +1,39 @@
+import type { User } from '@prisma/client';
+
+import dbClient from '../config/database.server';
+import type { UserProfile } from '../types';
+
+class UserRepository {
+  static async findBy(param: UserProfile): Promise<User> {
+    try {
+      return (await dbClient.user.findUnique({
+        where: param,
+      })) as User;
+    } catch (err) {
```

Comment:
I think we can remove the `try` + `catch` + `throw` here, since the catch only re-throw the error 🤔

Reply:
The same may applies to the `updateOrCreate` 👀

Reply:
Can you a bit more about re-throw ?, because in my app in `root.tsx` has Error boundary that I try catch to throw the error to call the Error boundary.

Reply:
Hmm 🤔 Let me take a look at this

Reply:
What I meant in the above just it's that instead of doing this:

```typescript
static async findBy(param: UserProfile): Promise<User> {
  try {
    return (await dbClient.user.findUnique({
      where: param,
    })) as User;
  } catch (err) {
    throw err;
  }
}
```

We can instead make it more concise like this 😆:
```typescript
static async findBy(param: UserProfile): Promise<User> {
  return (await dbClient.user.findUnique({
    where: param,
  })) as User;
}
```

Does doing this impact the behavior of your Error boundary? 🤔

Reply:
Oh, now I understand what you mean re-throw, actually we don't need to use `try` +`catch` +`throw` because `prisma` it throw error by itself. Thanks for trying to help me on this 😅🙏🏼. (I will do the change)

Reply:
• [65182a7](https://github.com/nimblehq/js-framework-benchmark/pull/91/commits/65182a70a4c14597e4027c744c4ccaaadd4132f3) Fixed here.